pkg/injector/pod_patch.go

```diff
@@ -759,21 +793,35 @@ func getSidecarContainer(annotations map[string]string, id, daprSidecarImage, im
 		},
 		ReadinessProbe: &corev1.Probe{
 			ProbeHandler:        httpHandler,
-			InitialDelaySeconds: getInt32AnnotationOrDefault(annotations, daprReadinessProbeDelayKey, defaultHealthzProbeDelaySeconds),
-			TimeoutSeconds:      getInt32AnnotationOrDefault(annotations, daprReadinessProbeTimeoutKey, defaultHealthzProbeTimeoutSeconds),
-			PeriodSeconds:       getInt32AnnotationOrDefault(annotations, daprReadinessProbePeriodKey, defaultHealthzProbePeriodSeconds),
-			FailureThreshold:    getInt32AnnotationOrDefault(annotations, daprReadinessProbeThresholdKey, defaultHealthzProbeThreshold),
+			InitialDelaySeconds: getInt32AnnotationOrDefault(cfg.annotations, daprReadinessProbeDelayKey, defaultHealthzProbeDelaySeconds),
+			TimeoutSeconds:      getInt32AnnotationOrDefault(cfg.annotations, daprReadinessProbeTimeoutKey, defaultHealthzProbeTimeoutSeconds),
+			PeriodSeconds:       getInt32AnnotationOrDefault(cfg.annotations, daprReadinessProbePeriodKey, defaultHealthzProbePeriodSeconds),
+			FailureThreshold:    getInt32AnnotationOrDefault(cfg.annotations, daprReadinessProbeThresholdKey, defaultHealthzProbeThreshold),
 		},
 		LivenessProbe: &corev1.Probe{
 			ProbeHandler:        httpHandler,
-			InitialDelaySeconds: getInt32AnnotationOrDefault(annotations, daprLivenessProbeDelayKey, defaultHealthzProbeDelaySeconds),
-			TimeoutSeconds:      getInt32AnnotationOrDefault(annotations, daprLivenessProbeTimeoutKey, defaultHealthzProbeTimeoutSeconds),
-			PeriodSeconds:       getInt32AnnotationOrDefault(annotations, daprLivenessProbePeriodKey, defaultHealthzProbePeriodSeconds),
-			FailureThreshold:    getInt32AnnotationOrDefault(annotations, daprLivenessProbeThresholdKey, defaultHealthzProbeThreshold),
+			InitialDelaySeconds: getInt32AnnotationOrDefault(cfg.annotations, daprLivenessProbeDelayKey, defaultHealthzProbeDelaySeconds),
+			TimeoutSeconds:      getInt32AnnotationOrDefault(cfg.annotations, daprLivenessProbeTimeoutKey, defaultHealthzProbeTimeoutSeconds),
+			PeriodSeconds:       getInt32AnnotationOrDefault(cfg.annotations, daprLivenessProbePeriodKey, defaultHealthzProbePeriodSeconds),
+			FailureThreshold:    getInt32AnnotationOrDefault(cfg.annotations, daprLivenessProbeThresholdKey, defaultHealthzProbeThreshold),
 		},
 	}
 
-	c.Env = append(c.Env, utils.ParseEnvString(annotations[daprEnvKey])...)
+	// This is a special case for virtual kubelets on Azure Container Instances (ACI).
+	// ACI does not support separate Command and Args,
+	// Ref. https://github.com/virtual-kubelet/azure-aci/issues/10.
+	// For other deployments, the Command is passed as a part of Args.
+	// This allows the Docker images to specify an ENTRYPOINT
+	// which is otherwise overridden by Command.
+	if isACIVirtualKubelet(cfg.tolerations) {
```

Comment:
Please remove all and every reference to ACI preferably to Virtual Kubelet also from the code. Please make these settings configurable on the Helm Chart and have the code act based on the configured settings.

Reply:
Updated.